processing/src/main/java/org/apache/druid/segment/serde/DictionaryEncodedStringIndexSupplier.java

```diff
@@ -77,42 +78,45 @@ public DictionaryEncodedStringIndexSupplier(
     this.dictionaryUtf8 = dictionaryUtf8;
     this.bitmaps = bitmaps;
     this.indexedTree = indexedTree;
+    this.zeroIsNull = NullHandling.isNullOrEquivalent(dictionary.get(0));
```

Comment:
Will this materially slow down historical startup time? If so: lazily compute instead of eagerly (like `minTime`, `maxTime` in QueryableIndexStorageAdapter).

Reply:
it didn't seem that expensive to just do this when we need it, so i just moved this into `as` method